server/consumer.go

```diff
@@ -2271,10 +2271,27 @@ func configsEqualSansDelivery(a, b ConsumerConfig) bool {
 }
 
 // Helper to send a reply to an ack.
-func (o *consumer) sendAckReply(subj string) {
+func (o *consumer) sendAckReply(subj string, err *ApiError) {
 	o.mu.RLock()
 	defer o.mu.RUnlock()
-	o.outq.sendMsg(subj, nil)
+	o.sendAckReplyLockHeld(subj, err)
+}
+
+// Helper to send a reply to an ack.
+// Lock must be held
+func (o *consumer) sendAckReplyLockHeld(subj string, err *ApiError) {
```

Comment:
Usually functions that require a lock to be held end with `Locked`

Reply:
Ok changed.